### PR TITLE
fix ci

### PR DESCRIPTION
**Pull request by graingert-coef**
## About
CI was failing, so I fixed the failing tests. After this I found that the tests were flaky in starting up the browsers for testing so I upgraded to the new cimg images with updated browsers.

## Pre-Merge checklist
- [ ] The project was correctly built with `npm run build`.
- [ ] If there was any conflict, it was solved correctly.
- [ ] All changes were documented in CHANGELOG.md.
- [x] All tests on CircleCI have passed.
- [ ] All Percy visual changes have been approved.
- [ ] Two people have :dancer:'d the pull request. You can be one of these people if you are a Dash Deck core contributor.

## Reference Issues

Closes #[issue number]

## Other comments

### CODE REVIEW

**Comment by graingert-coef** [comment]
@jingningzhang1 can I get a review please?

File: tests/requirements.txt [hunk 1]
```diff
@@ -4,4 +4,5 @@
 cryptography<3.4;python_version<"3.7"
 pyproj~=2.2
 dash[dev,testing]>=1.3.1
-Pillow
\ No newline at end of file
+Pillow
+black
```
Comment (alexcjohnson) [inline tests/requirements.txt:8]: Should specify an exact version of `black` to minimize the tests breaking on their own.

**Comment by alexcjohnson** [comment]
@graingert-coef this looks great, thanks! Is this a precursor to #22 or are they independent?

FYI the stance we've taken throughout the Dash ecosystem is to drop Py3.7 support at this point as well, so 3.8 will be the minimum supported version in the next release of any package.

**Comment by graingert-coef** [comment]
@alexcjohnson this is a precursor to #22

File: tests/requirements.txt (hunk 1 above)
Comment (graingert-coef) [inline tests/requirements.txt:8]:
```suggestion
black~=23.12
```

File: tests/requirements.txt [hunk 2]
```diff
@@ -4,4 +4,5 @@
 cryptography<3.4;python_version<"3.7"
 pyproj~=2.2
 dash[dev,testing]>=1.3.1
-Pillow
\ No newline at end of file
+Pillow
+black~=23.12
```
Comment (graingert-coef) [inline tests/requirements.txt:8]:
```suggestion
black~=23.3
```

File: tests/requirements.txt [hunk 3]
```diff
@@ -4,4 +4,5 @@
 cryptography<3.4;python_version<"3.7"
 pyproj~=2.2
 dash[dev,testing]>=1.3.1
-Pillow
\ No newline at end of file
+Pillow
+black~=23.3
```
Comment (graingert-coef) [inline tests/requirements.txt:8]:
```suggestion
black~=22.8
```

**Comment by graingert-coef** [comment]
> FYI the stance we've taken throughout the Dash ecosystem is to drop Py3.7 support at this point as well, so 3.8 will be the minimum supported version in the next release of any package.

I've dropped Python 3.6 and 3.7 in #22

File: .circleci/config.yml [hunk 4]
```diff
@@ -61,29 +61,29 @@ jobs:
...
 
             - restore_cache:
-                key: deps1-{{ .Branch }}-{{ checksum "tests/requirements.txt" }}-{{ checksum "demos/requirements.txt" }}-{{ checksum "requirements.txt" }}
+                key: deps2-{{ .Branch }}-{{ checksum "tests/requirements.txt" }}-{{ checksum "demos/requirements.txt" }}-{{ checksum "requirements.txt" }}
 
             - run:
                 name: Create virtualenv
                 command: |
-                    python -m venv venv
+                    python -m venv dash-deck-venv
```
Comment (alexcjohnson) [inline .circleci/config.yml:69]: Is that true, these `cimg/python` images have a default `venv` that conflicts with the one we're making? Not in some parent directory? That's extremely unfortunate if true, and will have an impact all across Dash.

**Review by alexcjohnson — APPROVED** [review]
💃 @graingert-coef Nice job, happy to merge and move on to reviewing #22 but I would like to understand the default `venv` change as that could have very wide implications - do you have a reference for that, or can you explain how you determined that `cimg/python` is already using `venv`?

**Comment by graingert-coef** [comment]
failing CI was here: https://app.circleci.com/pipelines/github/plotly/dash-deck/51/workflows/d5bd5a29-ab59-490c-a665-972497fee596/jobs/149?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link&utm_content=summary
I googled the error message and found this: https://discuss.circleci.com/t/python-venv-setup-no-longer-working-as-expected-build-failing/45531/2 and this https://discuss.circleci.com/t/upgrading-to-the-convenience-image-breaks-everything/42746

so seems just changing the cache key should fix it

File: .circleci/config.yml (hunk 4 above)
Comment (graingert-coef) [reply .circleci/config.yml]: it is not true, I was tricked by the error message

**Comment by alexcjohnson** [comment]
OK great, really appreciate you digging into the `venv` issue, due to its impact on other projects!

**Pull request merged**